src/shared/api/dto.ts

```diff
@@ -1,183 +1,228 @@
-/* eslint-disable no-unused-vars */
-/* eslint-disable camelcase */
-import type { JsonObject } from 'type-fest';
+import { z } from 'zod';
+import BigNumber from 'bignumber.js';
+import invariant from 'tiny-invariant';
+
+const paginationSchema = z.object({
```

Comment:
Tip: in TypeScript, there are _runtime_ and _type_ independent namespaces. So, you can have a runtime item (variable, function) and a type item to have the same name.

Therefore, you can have a `const Pagination` that is a Zod schema, and a `type Pagination` that is `z.infer<typeof Pagination>`.

I would strongly advocate to this rename because having `xxxSchema` for all schemas is kind of verbose.

```suggestion
const Pagination = z.object({
```

Reply:
Well, with such naming we can easy understand either it is a type or a zod object. Additionally, when I have
```
export type Account 
export const Account
```
and write `Account` typescript doesn't know what I want to import - type or schema. Hence, instead of auto import I have to check it's correctness every time.

The following might be useful:
```
export type Account
export const account
```

But I would keep it as is.

Reply:
> and write `Account` typescript doesn't know what I want to import - type or schema

TypeScript perfectly knows what you are referring to depending on _where_ you are referring to it - in runtime or in type level.

> Hence, instead of auto import I have to check it's correctness every time.

What's the problem? It just a single `import { Account }` for both cases.

I would insist on this uniform naming.

However, a problem could occur is with classes such as `AccountId`, because classes are both types of the instance and also a value of the constructor. In this case having a `Schema` suffix is fine, imo.